File: system-test/java/src/main/java/com/optivem/shop/dsl/port/given/steps/GivenPromotion.java
package com.optivem.shop.dsl.port.given.steps;

import com.optivem.shop.dsl.port.given.steps.base.GivenStep;

public interface GivenPromotion extends GivenStep {
    GivenPromotion withActive(boolean promotionActive);
    GivenPromotion withDiscount(double discount);
    GivenPromotion withDiscount(String discount);
}

File: system-test/typescript/src/dsl/scenario/given/given-promotion.ts
import { PromotionConfig } from '../scenario-context';
import { ThenContractStage } from '../then/then-contract';
import { WhenStage } from '../when/when-stage';
import type { GivenStage } from './given-stage';

export class GivenPromotion {
  constructor(
    private readonly stage: GivenStage,
    private readonly config: PromotionConfig,
  ) {}

  withActive(promotionActive: boolean): GivenPromotion {
    this.config.promotionActive = promotionActive;
    return this;
  }

  withDiscount(discount: number | string): GivenPromotion {
    this.config.discount = typeof discount === 'number' ? discount.toFixed(2) : discount;
    return this;
  }

  and(): GivenStage {
    return this.stage;
  }

  when(): WhenStage {
    return this.stage.when();
  }

  then(): ThenContractStage {
    return this.stage.then();
  }
}
